vms/main.tf

```diff
@@ -0,0 +1,36 @@
+variable "deployment_id" {}
+variable "instance_name_suffix" {}
+variable "vm_name_prefix" {}
+variable "latest_al2_linux_ami" {}
+
+variable "app_tag" {
+  description = "Tag value for AAP component"
+  validation {
+    condition = var.app_tag == "controller" || var.app_tag == "hub"
+    error_message = "Invalid app_tag. Valid values are 'controller' or 'hub'."
+  }
+  type = string
+  default = "controller"
+}
+
+resource "aws_instance" "aapvm" {
+  ami = var.latest_al2_linux_ami
+  instance_type = "m5a.xlarge"
```

Comment:
This might need to be a parameter too since the different types of instances would need to be different sizes and even within the same type, one user might want to use beefier machines than someone doing dev/test/trial

Reply:
added instance_type as a variable with default value `m5a.xlarge`